File: Pro_ODP_NET_for_Oracle_Database_11g-4541/vb_dot_net/Chapter5VBNET/UDTs/ProductTypeFactory.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Oracle.DataAccess.Types
Imports Oracle.DataAccess.Client

<OracleCustomTypeMappingAttribute("EDZEHOO.PRODUCTTYPE")> _
Public Class PRODUCTTYPEFactory
    Implements IOracleCustomTypeFactory

    Public Overridable Function CreateObject() As IOracleCustomType Implements Oracle.DataAccess.Types.IOracleCustomTypeFactory.CreateObject
        Dim obj As New PRODUCTTYPE()
        Return obj
    End Function 'CreateObject
End Class 'PRODUCTTYPEFactory

File: Pro_ODP_NET_for_Oracle_Database_11g-4541/c_sharp/Chapter 5/UDTs/ProductTypeFactory.cs
using System;
using System.Collections.Generic;
using System.Text;
using Oracle.DataAccess.Types;
using Oracle.DataAccess.Client;

[OracleCustomTypeMappingAttribute("EDZEHOO.PRODUCTTYPE")]
public class PRODUCTTYPEFactory : IOracleCustomTypeFactory
{
    public virtual IOracleCustomType CreateObject()
    {
        PRODUCTTYPE obj = new PRODUCTTYPE();
        return obj;
    }
}